ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/workspace/WorkspaceEventsHandler.java

```diff
@@ -399,8 +419,13 @@ protected void onMessageReceived(WorkspaceStatusEvent statusEvent) {
 
                 case STOPPED:
                     loader.setSuccess(LoaderPresenter.Phase.STOPPING_WORKSPACE);
-                    unSubscribeHandlers();
-                    eventBus.fireEvent(new WorkspaceStoppedEvent(workspace));
+
+                    try {
```

Comment:
So if it's really the general problem should we wrap every firing an exception to such try block in order to take care about all problems of all handlers?
I doubt it.
I think that all exceptions in handlers are the problems of those handlers. And code that fires some event shouldn't worry about those exceptions in handlers.

IMO think such wrapping would bring a bad practice to our code.
Anyway, I'm really interested in your opinion.